### PR TITLE
Hkh/webappdeploy

### PR DESCRIPTION
**Pull request by HosseinHeris**
# Description

Add assets and update readme for webapp deployment. No interferences with the rest of the code. Done tests by following the instructions and creating webapp and testing it using cosmos-mongo. 

## Type of change 

Please remove options that are not relevant. 
- [x] New feature: not breaking, new functionality 
- [] New feature: may cause existing code to not work as expected 
- [] Bug fix 
- [x] Needs documentation update 

## Test and checks 
Please describe how you've tested your file 
- [] tested via a notebook
- [x] tested via a testflow 
- [] only made sure the code is written correctly 
Checks
- [x] commented code for complex areas
- [x] updated readme with the new changes 
- [x] changes not generating errors with the main flows in rag-vectorsearchtype directories (e.g rag-cosmos-mongo)
- [x] have tested my implementation 
- [x] have updated environment.yaml file with any additional dependencies

### CODE REVIEW

File: financial_transcripts/deployment_utilities/webapp_deployment/deploy_via_azure_cli_instructions.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+      2. Create a container registry ` az acr create --name <ACR-NAME> --resource-group <RG-NAME> --sku basic --admin-enabled true`
+      3. Create a web app service plan `az appservice plan create --resource-group <RG-NAME> --name <APP-SERVICE-PLAN-NAME> --location eastus --is-linux --sku B2` (please make sure to to use B2 for promptflow deployments)
+   2. Build the docker container remotely (this will upload and build the container in your ACR service) `az acr build --registry <ACR-NAME> --resource-group <RG-NAME> --image <rag-app-image-name> .`
+   3. Create a web app using the built container `az webapp create --resource-group <RG-NAME> --plan <APP-SERVICE-PLAN-NAME> --name <RAG-WEBAPP-NAME> -i <ACR-NAME>.azurecr.io/<rag-app-image-name>:latest`
+   plase double check the deployment center section of the web app on the azure portal to ensure image is pulled correctly.  
+   4. Configure your web app to listen to port 8000 and set a longer container_start_time_limit `az webapp config appsettings set --resource-group <RG-NAME> --name <RAG-WEBAPP-NAME> --settings WEBSITES_PORT=8000 WEBSITES_CONTAINER_START_TIME_LIMIT=1800 `
```
Comment (journeyman-msft) [inline financial_transcripts/deployment_utilities/webapp_deployment/deploy_via_azure_cli_instructions.md:12]: please change to 8080 to be consistent with promptflow

File: financial_transcripts/deployment_utilities/load_testing/test_run_instructions.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+`pip install locust`
+Note: It is listed in the environment.yml file and should already be installed in the appliedaipf conda environment. 
+
+## usage
+1. Modify the file `backend_loadtest.py` if you want to modify the question or test new endpoints
+2. Run `locust -f backend_loadtest.py -u 3 -r 10 -t 3m --html report.html` in your conda terminal 
+    1. Add --headless to run headless
+    2. -u is the number of max concurrent users
+    3. -r is the rate to spawn users per second
+    4. -t is the time of the test (e.g. 1h, 1m, 60 (default unit is s)
```
Comment (journeyman-msft) [inline financial_transcripts/deployment_utilities/load_testing/test_run_instructions.md:19]: unclosed paranthesis

**Review by journeyman-msft — CHANGES_REQUESTED** [review]
I would suggest removing locust from the `environment.yaml`. The README with `test_run_instructions.md` says you can pip install there and that should be sufficient. Please upgrade to promptflow==1.7.0 . Here are the results from my deployed webapp

image

File: financial_transcripts/deployment_utilities/webapp_deployment/deploy_via_azure_cli_instructions.md (hunk 1 above)
Comment (HosseinHeris) [reply financial_transcripts/deployment_utilities/webapp_deployment/deploy_via_azure_cli_instructions.md]: done!

File: README.md
[the remark this answers is not in the document]
Comment (HosseinHeris) [reply README.md:1]: added all!

File: financial_transcripts/deployment_utilities/load_testing/test_run_instructions.md (hunk 2 above)
Comment (HosseinHeris) [reply financial_transcripts/deployment_utilities/load_testing/test_run_instructions.md]: added!

File: financial_transcripts/rag-cosmos-mongo/llm_jodge.jinja2
Comment (journeyman-msft) [inline financial_transcripts/rag-cosmos-mongo/llm_jodge.jinja2:1]: llm_jodge is the file name , is it supposed to be judge?

**Review by journeyman-msft — APPROVED** [review]
Thanks for addressing. Just a little nit pick, but not blocking : langchain version >= should be ==

**Comment by HosseinHeris** [comment]
removed the mistakenly added jinja file as well.

**Pull request merged**